src/actions/company.js

```diff
@@ -216,6 +219,53 @@ export const queryCompanyTimeAndSalary = ({
   }
 };
 
+const setTimeAndSalaryStatistics = (companyName, box) => ({
+  type: SET_TIME_AND_SALARY_STATISTICS,
+  companyName,
+  box,
+});
+
+export const queryCompanyTimeAndSalaryStatistics = ({ companyName }) => async (
+  dispatch,
+  getState,
+) => {
+  const box = companyTimeAndSalaryStatisticsBoxSelectorByName(companyName)(
+    getState(),
+  );
+  if (isFetching(box) || (isFetched(box) && box.data.name === companyName)) {
+    return;
+  }
+
+  dispatch(setTimeAndSalaryStatistics(companyName, toFetching()));
+
+  try {
+    const data = await getCompanyTimeAndSalaryStatistics({
+      companyName,
+    });
+
+    // Not found case
+    if (data == null) {
+      return dispatch(
+        setTimeAndSalaryStatistics(companyName, getFetched(data)),
+      );
+    }
+
+    const timeAndSalaryStatisticsData = {
+      name: data.name,
+      salary_work_time_statistics: data.salary_work_time_statistics,
+    };
```

Comment:
Is this intentional to show the data structure of redux state (so, we know there are two keys)

Reply:
Not quite, simply following other patterns. But an obvious upside is  like what you mentioned.